src/arrayMethodSort.js

```diff
@@ -5,8 +5,30 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    const length = this.length;
+
+    for (let i = 0; i < length - 1; i++) {
+      for (let j = i + 1; j < length; j++) {
+        if (compareFunction && typeof compareFunction === 'function') {
+          if (compareFunction(this[j], this[i]) < 0) {
+            const temp = this[i];
+
+            this[i] = this[j];
+            this[j] = temp;
+          }
+        } else {
+          if (`${this[j]}` < `${this[i]}`) {
+            const temp = this[i];
+
+            this[i] = this[j];
+            this[j] = temp;
+          }
+        }
```

Comment:
I suggest just create default compareFunction and use it if there is no callback was passed. Because now you have duplicated code on 14-17 and 21-24 lines
```suggestion
          if (compareFunction(this[j], this[i]) < 0) {
            const temp = this[i];

            this[i] = this[j];
            this[j] = temp;
          }
```